provisioner/terraform/tfparse/tfparse.go

```diff
@@ -0,0 +1,490 @@
+package tfparse
+
+import (
+	"archive/zip"
+	"bytes"
+	"context"
+	"encoding/json"
+	"io"
+	"os"
+	"slices"
+	"sort"
+	"strconv"
+	"strings"
+
+	"github.com/coder/coder/v2/archive"
+	"github.com/coder/coder/v2/provisionersdk"
+	"github.com/coder/coder/v2/provisionersdk/proto"
+
+	"github.com/hashicorp/hcl/v2"
+	"github.com/hashicorp/hcl/v2/hclparse"
+	"github.com/hashicorp/hcl/v2/hclsyntax"
+	"github.com/hashicorp/terraform-config-inspect/tfconfig"
+	"github.com/zclconf/go-cty/cty"
+	"golang.org/x/xerrors"
+
+	"cdr.dev/slog"
+)
+
+// NOTE: This is duplicated from coderd but we can't import it here without
+// introducing a circular dependency
+const maxFileSizeBytes = 10 * (10 << 20) // 10 MB
+
+// WorkspaceTags extracts tags from coder_workspace_tags data sources defined in module.
+// Note that this only returns the lexical values of the data source, and does not
+// evaluate variables and such. To do this, see evalProvisionerTags below.
+func WorkspaceTags(ctx context.Context, logger slog.Logger, module *tfconfig.Module) (map[string]string, error) {
+	workspaceTags := map[string]string{}
+
+	for _, dataResource := range module.DataResources {
+		if dataResource.Type != "coder_workspace_tags" {
+			logger.Debug(ctx, "skip resource as it is not a coder_workspace_tags", "resource_name", dataResource.Name, "resource_type", dataResource.Type)
```

Comment:
This log line feels very noisy. Do we need it? 

If we do, can we aggregate all the skips into a `map[string]string` or something, and consolidate into 1 log line outside the loop?

```
logger.Debug(ctx, "skiped resources as it is not a coder_workspace_tags", slog.F("skipped", map[string]string{
    dataResource.Name:dataResource.Type,
}))
```

Reply:
Let me comment as I'm the original author of the parsing logic. I used it mostly for debugging purposes, I doubt there is any point in exposing it to users. We can either implement your idea of aggregated errors or add another level  `logger.Trace()`.

Reply:
Trace/Aggregate work for me :+1: